src/internal/proc.js

```diff
@@ -244,6 +244,9 @@ export default function proc(
       iterator._deferredEnd && iterator._deferredEnd.resolve(result)
     } else {
       if(result instanceof Error) {
+        if(onerror) {
```

Comment:
Ok I will move it down. Do we need to check for `instanceof Error` in there as well then? What other kinds of `result` can we expect here?